app/models/user.rb

```diff
@@ -659,11 +659,11 @@ def grouped_unread_notifications
     results.to_h
   end
 
-  ###
-  # DEPRECATED: This is only maintained for backwards compat until v2.5. There
-  # may be inconsistencies with counts in the UI because of this, because unread
-  # high priority includes PMs AND bookmark reminders.
   def unread_private_messages
+    Discourse.deprecate(
+      "#unread_private_messages is deprecated, use #unread_high_priority_notifications instead.",
+      drop_from: "2.5.0",
```

Comment:
Put 2.5 here to preserve the metadata from the original comment. Actual drop will likely be in 3.3.